File: 05_arrays/05_15_majority_element/majority_element.js
const majorityElement = (nums = []) => {
    [count, element] = [1, nums[0]];

    for (let i = 1; i < nums.length; i++) {
        if (count === 0) {
            element = nums[i];
            count++;
        } else if (element === nums[i]) {
            count++;
        }  else {
            count--;
        }
    }

    return element;
}

const arr = [2, 2, 1, 1, 1, 2, 2];
console.log(`Elements: ${arr}`);
const ans = majorityElement(arr);
console.log("The majority element is:", ans);

File: 05_arrays/05_15_majority_element/MajorityElement.java
import java.util.Arrays;

public class MajorityElement {
    public static void main(String args[]) {
        int[] arr = {2, 2, 1, 1, 1, 2, 2};
        System.out.println("Elements: " + Arrays.toString(arr));
        int ans = majorityElement(arr);
        System.out.println("The majority element is: " + ans);

    }
    
    private static int majorityElement(int nums[]) {
        int count = 1;
        int element = nums[0];

        for (int i = 0; i < nums.length; i++) {
            if (count == 0) {
                count++;
                element = nums[i];
            } else if (element == nums[i]) {
                count++;
            } else {
                count--;
            }
        }

        return element;
    }
}
